cmd/contour/serve.go

```diff
@@ -76,6 +76,7 @@ func registerServe(app *kingpin.Application) (*kingpin.CmdClause, *serveContext)
 		}
 		defer f.Close()
 		dec := yaml.NewDecoder(f)
+		dec.SetStrict(true)
```

Comment:
I think that we should decorate the YAML error, since it doesn't give much context:
```
$ contour serve -c /tmp/t
contour: error: yaml: unmarshal errors:
  line 2: field configs not found in type main.serveContext, try --help
```

Suggest:
```Go
if err := dec.Decode(&ctx); err != nil {
        return fmt.Errorf("failed to parse configuration file %q: %w", configFile, err)
}

return nil
```

Reply:
I agree with decorating the error.

I just tested your suggestion.  This is the output. 

```
contour: error: failed to parse configuration file "/config/contour.yaml": yaml: unmarshal errors:
  line 54: field defaulst-http-versions not found in type main.serveContext, try --help
```

`configFile` might be a little misleading if the user used a configMap to configure contour. This could be enough. 
```
if err := dec.Decode(&ctx); err != nil {
        return fmt.Errorf("failed to parse contour configuration: %w", err)
}
return nil
```

If you want you could also be more verbose.

```
if err := dec.Decode(&ctx); err != nil {
        return fmt.Errorf("failed to parse contour configuration: %w Valid values can be found here: https://projectcontour.io/docs/%s/configuration/ ", err,version)
}
return nil
```

Reply:
I think the second version is fine with "failed to parse contour configuration" is fine, if you prefer that. Thanks Tim 👍

Reply:
I will add it. btw sorry for the mess here, I made a git mistake while trying to signoff an old commit.